common/views/components/ImageViewer/ImageViewer.js

```diff
@@ -90,6 +90,9 @@ const ViewerContent = ({
   return (
     <div className={`${classes} image-viewer__content image-viewer__content2`}>
       <div className="image-viewer__controls flex flex-end flex--v-center">
+        {/* Setting OpenSeaDragon's `showRotationControl` option to `true` means we have to provide
+        a dummy `rotateLeftButton` or handle rotation programatically. The former is simpler. */}
+        <span id={`rotate-left-${id}`} className={'is-hidden'} />
```

Comment:
How do we feel about this? To me it felt better than querying for the rotateRight button and adding a click handler to get/set rotation?

Reply:
We could start getting a little more programmatic about it and use the methods exposed on the `Viewer` / `Viewport`?

https://openseadragon.github.io/docs/OpenSeadragon.Viewport.html#setRotation

Reply:
Yeah, that's what I was wondering. Is there a more React-y way of doing this programmatically than e.g.

```
useEffect(() => {
    if (viewer) {
      viewer.destroy();
      setViewer(null);
    }

    setupViewer(infoUrl, id, handleScriptError).then(viewer => {
      const rotateButton = document.getElementById(`rotate-right-${id}`);

      rotateButton &&
        rotateButton.addEventListener('click', () => {
          viewer.viewport.setRotation(viewer.viewport.getRotation() + 90);
        });

      setViewer(viewer);
    });
  }, [infoUrl]);
```

?

Reply:
You could add the event to the button the normal way and have a check on it.
```JSX
{viewer &&
  <Control
    onClick={() => viewer.viewport.setRotation(
      viewer.viewport.getRotation() + 90
    )}} />
```